File: research/explorations/bench-realistic/fib.lua
local function fib(n)
    if n <= 1 then return n end
    return fib(n - 1) + fib(n - 2)
end

local N = tonumber(arg[1]) or 25
-- warmup
for i = 1, 100 do fib(N) end
local iters = 1000
local clock = os.clock
local start = clock()
local r
for i = 1, iters do r = fib(N) end
local elapsed = (clock() - start) * 1e9
io.write(string.format("result:     %d\n", r))
io.write(string.format("iterations: %d\n", iters))
io.write(string.format("total:      %.2fms\n", elapsed / 1e6))
io.write(string.format("per call:   %.0fns\n", elapsed / iters))

File: research/explorations/bench-realistic/fib.rs
use std::env;
use std::time::Instant;

fn fib(n: i64) -> i64 {
    if n <= 1 { return n; }
    fib(n - 1) + fib(n - 2)
}

fn main() {
    let n: i64 = env::args().nth(1).and_then(|s| s.parse().ok()).unwrap_or(25);
    // warmup
    for _ in 0..100 { let _ = fib(n); }
    let iters = 1000;
    let start = Instant::now();
    let mut r = 0i64;
    for _ in 0..iters { r = fib(n); }
    let elapsed = start.elapsed().as_nanos();
    println!("result:     {}", r);
    println!("iterations: {}", iters);
    println!("total:      {:.2}ms", elapsed as f64 / 1e6);
    println!("per call:   {}ns", elapsed / iters);
}
